lib/ansible/modules/remote_management/redfish/redfish_facts.py

```diff
@@ -0,0 +1,237 @@
+        elif command == "GetNicInventory":
+            result = rf_utils.get_nic_inventory()
+
+        # Storage
+        elif command == "GetStorageControllerInventory":
+            result = rf_utils.get_storage_controller_inventory()
+        elif command == "GetDiskInventory":
+            result = rf_utils.get_disk_inventory()
+
+        # Chassis
+        elif command == "GetFanInventory":
+            # execute only if we find Chassis resource
+            result = rf_utils._find_chassis_resource(rf_uri)
+            if result['ret'] is False:
+                module.fail_json(msg=result['msg'])
+            result = rf_utils.get_fan_inventory()
+
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    elif category == "Accounts":
+        # execute only if we find an Account service resource
+        result = rf_utils._find_accountservice_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        # Set default value
+        if not command:
+            command = "ListUsers"
+
+        if command == "ListUsers":
+            result = rf_utils.list_users()
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    elif category == "System":
+        # execute only if we find a System resource
+        result = rf_utils._find_systems_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        # Set default value
+        if not command:
+            command = "GetBiosAttributes"
+
+        if command == "GetBiosAttributes":
+            result = rf_utils.get_bios_attributes()
+        elif command == "GetBiosBootOrder":
+            result = rf_utils.get_bios_boot_order()
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    elif category == "Update":
+        # execute only if we find UpdateService resources
+        result = rf_utils._find_updateservice_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        # Set default value
+        if not command:
+            command = "GetFirmwareInventory"
+
+        if command == "GetFirmwareInventory":
+            result = rf_utils.get_firmware_inventory()
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    elif category == "Manager":
+        # execute only if we find a Manager service resource
+        result = rf_utils._find_managers_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        # Set default value
+        if not command:
+            command = "GetManagerAttributes"
+
+        if command == "GetManagerAttributes":
+            result = rf_utils.get_manager_attributes()
+        elif command == "GetLogs":
+            result = rf_utils.get_logs()
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    else:
```

Comment:
I didn't see a section or the Chassis category. It's listed as a valid choice in the module spec.

Reply:
Yup, good catch.